### ISSUE TITLE
Issues building learnyoumongo db.  gyp says it doesn't exist

### ISSUE THREAD

**Issue by kenisteward** [braz/learnyoumongodb#3]
Here are the errors.  they start with 2 learnyoumongodb is not found (although i'm trying to isntall it)

c:\Program Files\nodejs\node_modules\npm>npm install learnyoumongodb -g
\

> sleep@1.1.8 install C:\Users\kenneth\AppData\Roaming\npm\node_modules\learnyoumongodb\node_modules
> \sleep
> node build.js || nodejs build.js

child_process: customFds option is deprecated, use stdio instead.

C:\Users\kenneth\AppData\Roaming\npm\node_modules\learnyoumongodb\node_modules\sleep>if not defined
npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\....\node_mod
ules\node-gyp\bin\node-gyp.js" rebuild learnyoumongodb --global )  else (rebuild learnyoumongodb --g
lobal )
gyp: learnyoumongodb not found (cwd: C:\Users\kenneth\AppData\Roaming\npm\node_modules\learnyoumongo
db\node_modules\sleep) while trying to load learnyoumongodb
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\Program Files\nodejs\node_modules\npm\node_modules\n
ode-gyp\lib\configure.js:355:16)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Windows_NT 6.3.9600
gyp ERR! command "node" "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\
node-gyp.js" "rebuild" "learnyoumongodb" "--global"
gyp ERR! cwd C:\Users\kenneth\AppData\Roaming\npm\node_modules\learnyoumongodb\node_modules\sleep
gyp ERR! node -v v0.12.4
gyp ERR! node-gyp -v v2.0.1
gyp ERR! not ok
[sleep]: Error: Failed to execute 'node-gyp.cmd rebuild learnyoumongodb --global' (1)
child_process: customFds option is deprecated, use stdio instead.

C:\Users\kenneth\AppData\Roaming\npm\node_modules\learnyoumongodb\node_modules\sleep>if not defined
npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\bin\node-gyp-bin\....\node_mod
ules\node-gyp\bin\node-gyp.js" rebuild learnyoumongodb --global )  else (rebuild learnyoumongodb --g
lobal )
gyp: learnyoumongodb not found (cwd: C:\Users\kenneth\AppData\Roaming\npm\node_modules\learnyoumongo
db\node_modules\sleep) while trying to load learnyoumongodb
gyp ERR! configure error
gyp ERR! stack Error: `gyp` failed with exit code: 1
gyp ERR! stack     at ChildProcess.onCpExit (C:\Program Files\nodejs\node_modules\npm\node_modules\n
ode-gyp\lib\configure.js:355:16)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1074:12)
gyp ERR! System Windows_NT 6.3.9600
gyp ERR! command "node" "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\
node-gyp.js" "rebuild" "learnyoumongodb" "--global"
gyp ERR! cwd C:\Users\kenneth\AppData\Roaming\npm\node_modules\learnyoumongodb\node_modules\sleep
gyp ERR! node -v v0.12.4
gyp ERR! node-gyp -v v2.0.1
gyp ERR! not ok
[sleep]: Error: Failed to execute 'node-gyp.cmd rebuild learnyoumongodb --global' (1)
npm ERR! Windows_NT 6.3.9600
npm ERR! argv "C:\Program Files\nodejs\\node.exe" "C:\Program Files\nodejs\node_modules\npm\
\bin\npm-cli.js" "install" "learnyoumongodb" "-g"
npm ERR! node v0.12.4
npm ERR! npm  v2.10.1
npm ERR! code ELIFECYCLE

npm ERR! sleep@1.1.8 install: `node build.js || nodejs build.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the sleep@1.1.8 install script 'node build.js || nodejs build.js'.
npm ERR! This is most likely a problem with the sleep package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node build.js || nodejs build.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls sleep
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     c:\Program Files\nodejs\node_modules\npm\npm-debug.log

Thanks in advance!

**Comment by braz** [comment]
This appears to be related to the sleep module rather than the learnyoumongodb. "ERR! This is most likely a problem with the sleep package".
Do you have node or nodejs in your PATH?
If adding it to your PATH doesn't resolve it, please check with that module's maintainer (https://github.com/ErikDubbelboer/node-sleep).

**Comment by kenisteward** [comment]
I have gone through many posts already trying to figure this out.

I have Python 27 installed
I have the C++ compiler installed corretly for windows.
I have VS 2013
I have everything in path (python, nodejs, mongodb etc.)
I've run sleep and node-gyp by themselves perfectly fine.  It's not untill I try to install this module (plus aggregate adventure and the other one ) does this happen.

To use I've resulted in just downloading your module and running it by hand on node (with no path cause I can't))   

I actually posted in the sleep issues already.  He thought it was dev environment as well but nope it's not. (Unless I'm missing something)

I'm still looking for answers though .
